assets/javascripts/discourse/pre-initializers/chat-plugin-api.js

```diff
@@ -12,15 +13,21 @@ export default {
     withPluginApi("1.1.0", (api) => {
       const apiPrototype = Object.getPrototypeOf(api);
 
-      if (apiPrototype.hasOwnProperty("decorateChatMessage")) {
-        return;
+      if (!apiPrototype.hasOwnProperty("decorateChatMessage")) {
+        Object.defineProperty(apiPrototype, "decorateChatMessage", {
+          value(decorator) {
+            addChatMessageDecorator(decorator);
+          },
+        });
       }
 
-      Object.defineProperty(apiPrototype, "decorateChatMessage", {
-        value(decorator) {
-          addChatMessageDecorator(decorator);
-        },
-      });
+      if (!apiPrototype.hasOwnProperty("registerChatComposerButton")) {
```

Comment:
> * if the chat plugin isn't installed, calling `registerChatComposerButton` will fail, correct? Does it fail gracefully?

No it doesn’t atm, this is author's responsibility to handle this. We could improve it in core though to handle gracefully an unknown API call, that's a good idea, I will make a TODO out of it. :+1:

> extending the API from a plugin also probably means that we're adding new features to the API outside of its versioning system. As in, registerChatComposerButton would be available based on the chat plugin, but regardless of the API's version, which feels odd. (I don't know how to improve this, to be clear.)

Yes I don't have a good solution to this too, although it's wrapped in a `withPluginApi` block meaning  it should at least mean that's it's available since this API version.